native/src/main/java/io/ballerina/stdlib/http/api/client/actions/HttpClientAction.java

```diff
@@ -293,22 +296,66 @@ private static String[] getPathStringArray(BArray pathArray) {
 
     private static String constructQueryString(BMap params) {
         List<String> queryParams = new ArrayList<>();
+        Map<String, String> annotationValues = getQueryNameMapping(params);
         BString[] keys = (BString[]) params.getKeys();
         if (keys.length == 0) {
             return "";
         }
         for (BString key : keys) {
             Object value = params.get(key);
+            String queryName = key.getValue();
+            queryName = annotationValues.getOrDefault(queryName, queryName);
             String valueString = value.toString();
             if (value instanceof BArray) {
                 valueString = valueString.substring(1, valueString.length() - 1);
                 valueString = valueString.replace(QUOTATION_MARK, EMPTY);
             }
-            queryParams.add(key.getValue() + EQUAL_SIGN + valueString);
+            queryParams.add(queryName + EQUAL_SIGN + valueString);
         }
         return String.join(AND_SIGN, queryParams);
     }
 
+    /**
+     * This util function extracts the query name with the query annotation.
+     *
+     * @param params - Parameter map
+     * @return Map of string with overridden query param names
+     */
+    private static Map<String, String> getQueryNameMapping(BMap params) {
+        Map<String, String> annotationValues = new HashMap<>();
+        if (!(params.getType() instanceof RecordType queryRecord)) {
+            return annotationValues;
+        }
+        BMap<BString, Object> annotations = queryRecord.getAnnotations();
+
+        for (Map.Entry<BString, Object> annotation: annotations.entrySet()) {
+            BMap value = (BMap) annotation.getValue();
+            Object[] keys = value.getKeys();
+            for (Object annotRef: keys) {
+                String refRegex = "^ballerina/http:\\d+:(Query)$";
+                Pattern pattern = Pattern.compile(refRegex);
+                Matcher matcher = pattern.matcher(annotRef.toString());
+                if (matcher.find()) {
+                    extractedFieldName(annotationValues, annotation, value);
+                }
+            }
+        }
+        return annotationValues;
+    }
+
+    private static void extractedFieldName(Map<String, String> annotationValues, Map.Entry<BString, Object> annotation,
+                                           BMap value) {
+        for (Object object : value.values()) {
+            //store query parameters real name : $field$.<query name>
+            String regex = "(\\$field\\$\\.)";
+            String[] parts = Pattern.compile(regex).split(annotation.getKey().getValue());
+            String key = parts[1];
+            BMap bMap = (BMap) object;
+            BString overrideName = (BString) bMap.get(StringUtils.fromString("name"));
```

Comment:
Since the value is optional, better to check whether the value is `BString` before casting. Even though it is not 80% case, you can add a `nil` value for the name:
```bal
public type QueryParams record {|
    @http:Query {name: ()}
    string firstName;
|};
```

Reply:
Added the type check before casting, and curious if it allows a `()` value in the compilation although the field mentions having `string`  instead of `string?`.

Reply:
If the user provides an empty name value, such as `()` or `""`, the current implementation will fall back to the field name. Should we add an error or warning message prompting the user to provide a valid name, or should we keep the current behavior where an error response is returned?

Reply:
+1 to fallback the field name. There is no need for error or warning since this is not 80% case

Reply:
address via https://github.com/ballerina-platform/module-ballerina-http/pull/2143/commits/1218d3d373e183282f02752cd48815992384c3f6 and this for keeping the field name when the name is absent https://github.com/ballerina-platform/module-ballerina-http/blob/1218d3d373e183282f02752cd48815992384c3f6/native/src/main/java/io/ballerina/stdlib/http/api/client/actions/HttpClientAction.java#L311